### PR TITLE
Update sire extension

### PR DESCRIPTION
**Pull request by kandre**
Here is some corrections to the SIRE export extension. It complies better with SIRE requirements, especially with the Canton de Vaud's checker. Eventually  SIGE data pass trough the checker without errors with these scripts.

Some corrections were made based on errors coming from from SIGE dataset. It is not impossible that others datasets (built from different editing processes) may still have errors with Vaud's checker. Let's share our views !

Finally, those scripts can still be improved as there is still bunches of fields that are preset with static values.

### CODE REVIEW

**Comment by ponceta** [comment]
Travis down! Do we have a command for test reload?

Err:125 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease
  Could not connect to apt.postgresql.org:80 (72.32.157.246), connection timed out

**Comment by tudorbarascu** [comment]
The original author can just close and open again the pull request to restart travis.

**Comment by haubourg** [comment]
I just pushed on the button `restart build`.

**Comment by ponceta** [comment]
> I just pushed on the button `restart build`.

@haubourg You have some special magic buttons, reserved for great wizards!

Travis has gone green, although he seems a bit yellow on git. :  #1276 passed

File: extensions/ch_vd_sire/views/captage.sql [hunk 1]
```diff
@@ -1,32 +1,30 @@
 CREATE VIEW qwat_ch_vd_sire.captage AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/captage.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/conduite.sql [hunk 2]
```diff
@@ -1,29 +1,38 @@
 CREATE VIEW qwat_ch_vd_sire.conduite AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/conduite.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/regulation_pression.sql [hunk 3]
```diff
@@ -1,57 +1,30 @@
-CREATE VIEW qwat_ch_vd_sire.regulation_pression AS
+CREATE VIEW qwat_ch_vd_sire.regulation_pression AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/regulation_pression.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/reservoir.sql [hunk 4]
```diff
@@ -1,30 +1,40 @@
 CREATE VIEW qwat_ch_vd_sire.reservoir AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/reservoir.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/station_pompage.sql [hunk 5]
```diff
@@ -1,26 +1,24 @@
 CREATE VIEW qwat_ch_vd_sire.station_pompage AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/station_pompage.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/station_traitement.sql [hunk 6]
```diff
@@ -1,31 +1,56 @@
 CREATE VIEW qwat_ch_vd_sire.station_traitement AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/station_traitement.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/vanne_clapet.sql [hunk 7]
```diff
@@ -1,44 +1,37 @@
 CREATE VIEW qwat_ch_vd_sire.vanne_clapet AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/vanne_clapet.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/zone_pression.sql [hunk 8]
```diff
@@ -0,0 +1,12 @@
+CREATE VIEW qwat_ch_vd_sire.zone_pression AS
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/zone_pression.sql]: CREATE OR REPLACE
for more usability

File: extensions/ch_vd_sire/views/captage.sql [hunk 9]
```diff
@@ -1,32 +1,30 @@
...
-		, ST_Force2D(geometry) AS geometry
+		END AS "Chambre_De_Rassemblement",
+		altitude AS "Altitude",
+		qwat_ext_ch_vd_sire_adesafecter AS "A_Desaffecter_PDDE",
+		st_force2d(geometry) AS geometry
 	FROM qwat_od.vw_export_installation
-	WHERE
-		installation_type = 'source'
-		;
+	WHERE installation_type = 'source'::qwat_od.installation_type AND (status_active IS TRUE OR fk_status = 1306) AND fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/captage.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/conduite.sql [hunk 10]
```diff
@@ -1,29 +1,38 @@
...
+		NULL::double precision AS "Rugosite_Hydraulique",
+		st_length(geometry) AS "Longueur_Reelle",
+		2 AS "Calcul_Hydraulique",
+		qwat_ext_ch_vd_sire_adesafecter AS "A_Desaffecter_PDDE",
+		st_length(st_force2d(geometry)) AS "Shape_Length",
+		st_force2d(geometry) AS geometry
 	FROM qwat_od.vw_export_pipe
-	WHERE
-		pipe_function_major IS TRUE;
+	WHERE pipe_function_major IS TRUE AND (status_functional IS TRUE OR fk_status = 1306) AND fk_distributor = 1 AND NOT (pipe_function_code_sire = 0 AND st_length(geometry) < 10::double precision);
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/conduite.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/regulation_pression.sql [hunk 11]
```diff
@@ -1,57 +1,30 @@
...
-		, 0 AS Type_Regulation
-		, NULL AS Valeur_Consigne_Reg
-		, 2 AS Telecommande_Incendie
-		, 0 AS Centrale_Telecommande
-		, qwat_ext_ch_vd_sire_adesafecter AS A_Desaffecter_PDDE
-    , ST_Force2D(geometry) AS geometry
-	FROM qwat_od.vw_export_valve
-	WHERE fk_valve_function = 6101  -- vanne régulation
-		;
+	WHERE (fk_pressurecontrol_type = ANY (ARRAY[2801, 2802])) AND (status_functional IS TRUE OR fk_status = 1306) AND fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/regulation_pression.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/reservoir.sql [hunk 12]
```diff
@@ -1,30 +1,40 @@
...
+			ELSE 0
+		END AS "Telecommande_Incendie",
+		0 AS "Centrale_Telecommande",
+		qwat_ext_ch_vd_sire_adesafecter AS "A_Desaffecter_PDDE",
+		st_force2d(geometry) AS geometry
 	FROM qwat_od.vw_export_installation
-	WHERE
-		installation_type = 'tank'
-		;
+	WHERE installation_type = 'tank'::qwat_od.installation_type AND (status_functional IS TRUE OR fk_status = 1306) AND fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/reservoir.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/station_pompage.sql [hunk 13]
```diff
@@ -1,26 +1,24 @@
...
+		no_pumps AS "Nombre_Pompes",
+		manometric_height AS "H_Manometrique",
+		pump_operating_code_sire AS "Fonctionnement",
+		qwat_ext_ch_vd_sire_adesafecter AS "A_Desaffecter_PDDE",
+		st_force2d(geometry) AS geometry
 	FROM qwat_od.vw_export_installation
-	WHERE
-		installation_type = 'pump'
-		;
+	WHERE installation_type = 'pump'::qwat_od.installation_type AND (status_functional IS TRUE OR fk_status = 1306) AND fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/station_pompage.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/station_traitement.sql [hunk 14]
```diff
@@ -1,31 +1,56 @@
...
+		END AS "Decantation",
+		treatment_capacity AS "Capacite_Traitement",
+		2 AS "Electricite_Fonctionnement",
+		qwat_ext_ch_vd_sire_adesafecter AS "A_Desaffecter_PDDE",
+		st_force2d(geometry) AS geometry
 	FROM qwat_od.vw_export_installation
-	WHERE
-		installation_type = 'treatment'
-		;
+	WHERE installation_type = 'treatment'::qwat_od.installation_type AND (status_functional IS TRUE OR fk_status = 1306) AND fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/station_traitement.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/vanne_clapet.sql [hunk 15]
```diff
@@ -1,44 +1,37 @@
...
 	FROM qwat_od.vw_export_valve
-	WHERE
-		(
-			networkseparation IS TRUE
-			OR
-			fk_valve_function = 6107  -- vanne incendie
-			OR
-			fk_valve_type = 6307  -- clapet anti-retour
-		);
+	WHERE networkseparation IS TRUE OR fk_valve_function = 6107 OR fk_valve_type = 6307 AND (status_functional IS TRUE OR fk_status = 1306) AND fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/vanne_clapet.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

File: extensions/ch_vd_sire/views/zone_pression.sql [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+		pressurezone.id AS "ID_Num",
+		NULL::text AS "Remarque",
+		pressurezone.name AS "Nom_Descriptif",
+		distributor.name AS "Nom_Distributeur",
+		consumptionzone.qwat_ext_ch_vd_sire_pressure_load_type AS "Mise_En_Charge",
+		consumptionzone.qwat_ext_ch_vd_sire_pressure_load_altitude AS "Altitude_Plan_De_Charge"
+		FROM qwat_od.pressurezone
+		JOIN qwat_od.distributor ON pressurezone.fk_distributor = distributor.id
+		JOIN qwat_od.consumptionzone ON pressurezone.fk_consumptionzone = consumptionzone.id
+	WHERE pressurezone.fk_distributor = 1;
```
Comment (ponceta) [inline extensions/ch_vd_sire/views/zone_pression.sql]: +GRANT SELECT, REFERENCES, TRIGGER ON TABLE qwat_ch_vd_sire.captage TO qwat_viewer;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_user;
+GRANT ALL ON TABLE qwat_ch_vd_sire.captage TO qwat_manager;

(add Grants)

**Comment by kandre** [comment]
@ponceta I realize that this is PR is not yet merged, I'll handle it in the next days. Thanks for the review

**Comment by kandre** [comment]
Ready for review or merge. Anyone against rewriting this history ?

**Comment by lbartoletti** [comment]
LGTM @kandre 
@ponceta is it ok for you?

**Comment by lbartoletti** [comment]
@ponceta @kandre Since I'll start to work on the extension process, I'll merge it this week and after I'll move the extension in a separated project (qwat/extension_sire)

**Comment by kandre** [comment]
Thanks !

**Pull request merged**